pym/_emerge/resolver/package_tracker.py

```diff
@@ -31,41 +31,94 @@ def __len__(self):
+	to-be-installed package that was "replacing" an installed package, the installed package will "reappear". This
+	removal functionality is used by the slot conflict code in ``depgraph.py`` to modify the list of to-be-installed
+	packages as it addresses slot conflicts.
+
+	One of the main purposes of the PackageTracker is to detect conflicts between packages. Conflicts are detected
+	on to-be-installed packages only.
+
+	A slot conflict is a situation where a to-be-installed package is added to the package tracker via ``add_pkg()``,
+	and there is already a to-be-installed package added that has the same root, catpkg and slot. These cannot co-exist.
+
+	A cpv conflict is a situation where a to-be-installed package is added to the package tracker via ``add_pkg()``, and
+	there is already a to-be-installed package add that has the same root, catpkg, and version+revision. These cannot
+	co-exist.
+
+    The package tracker does not prevent slot and cpv conflicts from occurring. Instead, it allows them to be recorded
+    and the ``conflicts()`` and ``slot_conflicts()`` method will cause the package tracker to look at its internal data
+    structures and generate ``PackageConflict()`` objects for each conflict it finds.
+
+    The ``match()`` method is used extensively by ``depgraph.py`` to find packages that match a particular dependency
+    atom. The code now also supports soname dependencies
+
+	Future Functionality
+	====================
+
+	The package tracker may be extended in the future to track additional useful information:
+
+	* Packages that block one another. This information is not currently injected into the package tracker.
+
+	* Sub-slot conflicts. It is possible to identify situations where a to-be-installed package is in a new sub-slot.
+	  In this case, the depgraph can be queried for parents of this dependency, and these parents can be scheduled
+	  to be rebuilt.
 
-	Possible conflicts:
-		1) Packages that share the same SLOT.
-		2) Packages with the same cpv.
-		Not yet implemented:
-		3) Packages that block each other.
 	"""
 
 	def __init__(self, soname_deps=False):
+
 		"""
-		@param soname_deps: enable soname match support
-		@type soname_deps: bool
+		:param soname_deps bool: Determines whether support for soname deps should be enabled or not.
+		:var _cp_pkg_map: The collection of to-be-installed (not yet merged) packages. We care about conflicts in these
+		 packages.
+		:var _cp_vdb_pkg_map: The collection of already-installed packages.
+		:var _multi_pkgs: A list of keys in ``self._cp_pkg_map`` that have potential slot and cpv conflicts.
+		:var _replacing: The mechanism by which ``PackageTracker`` records to-be-installed packages that 'cover up'
+		 already-installed packages. ``self._replacing[cp_key] = [ new_pkg_that_replaced_cp_key... ]``.
+		:var _replaced_by: ``self.replaced_by[cp_key] == [ replaced_pkg_1, replaced_pkg_2 ]
```

Comment:
The instance attributes should be documented as shown in the example here:

http://www.sphinx-doc.org/en/stable/ext/autodoc.html#directive-autoattribute

Reply:
Maybe [info fields](http://www.sphinx-doc.org/en/stable/domains.html#info-field-lists) are more appropriate, as suggested here:

https://stackoverflow.com/a/8659919

Reply:
I converted them to info fields in 8fba660fb08ddd468d934c7478dcfc8015330e9b.